# HG changeset patch
# User Rik <rik@octave.org>
# Date 1363877104 25200
#      Thu Mar 21 07:45:04 2013 -0700
# Node ID ca395af72d8871804f86325b86f122736b8e79b4
# Parent  b8a4f313e78e016f6ba6966546567042992683e8
hsv.m: Make colormap compatible with Matlab (bug #36473).

* scripts/image/hsv.m: Make colormap compatible with Matlab (bug #36473).

diff --git a/scripts/image/hsv.m b/scripts/image/hsv.m
--- a/scripts/image/hsv.m
+++ b/scripts/image/hsv.m
@@ -18,17 +18,17 @@
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {} hsv (@var{n})
 ## Create color colormap.  This colormap begins with red, changes through
 ## yellow, green, cyan, blue, and magenta, before returning to red.
 ## It is useful for displaying periodic functions.  The map is obtained by
 ## linearly varying the hue through all possible values while keeping constant
 ## maximum saturation and value.  The equivalent code is
-## @code{hsv2rgb ([linspace(0,1,N)', ones(N,2)])}.
+## @code{hsv2rgb ([(0:N-1)'/N, ones(N,2)])}.
 ##
 ## The argument @var{n} must be a scalar.
 ## If unspecified, the length of the current colormap, or 64, is used.
 ## @seealso{colormap}
 ## @end deftypefn
 
 ## Author:  Kai Habel <kai.habel@gmx.de>
 
@@ -45,17 +45,17 @@ function map = hsv (n)
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 0, 0];
   elseif (n > 1)
-    hue = linspace (0, 1, n)';
+    hue = [0:n-1]' / n;
     map = hsv2rgb ([hue, ones(n,1), ones(n,1)]);
   else
     map = zeros (0, 3);
   endif
 
 endfunction
 
 
